mitodl/mit-open
# Shanbady/consolidate static fixtures

### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/4905
Closes https://github.com/mitodl/hq/issues/5145

### Description (What does it do?)
This PR solves the above 2 tickets related to required static data not being loaded in on fresh environments.
It does the following:
1. restores the running of data migrations in local environments (which i think was accidentally removed at some point)
2. moves the static fixture files to the data_fixtures app
3. moves the utility methods for loading static fixtures to the data_fixtures app
5. adds a method in the very initial data_fixtures data migration that manually loads static fixtures in before anything else
6. fixed a few other places where migrations could fail when using get_or_create/update without specifying defaults
7. updated the codespace's devcontainer.json to backpopulate resource channels after loading everything up

### How can this be tested?
There are two ways to try this - 
- #### in codespaces
    - spin up a codespace on this PR - make sure you spin it up via "new with options" and specify "4 core" before launching 
    <img width="283" alt="Screenshot 2024-10-11 at 10 26 07 AM" src="https://github.com/user-attachments/assets/d944dc31-522c-44d2-86c3-e77a91ff1fc5">
    <img width="521" alt="Screenshot 2024-10-11 at 10 26 17 AM" src="https://github.com/user-attachments/assets/77a464bf-c0ed-4e4c-b6f8-83583c53145f">

    - once the codespace is up (make sure you make port 8603 public) you should see topics show up on the channel page and there should be search data
- #### locally
    - do a `docker compose down` and `docker compose up`
    - all migrations should go through without issues
    - you should also be able to run the presviously failing commands referenced [here](https://github.com/mitodl/hq/issues/5145) with no issue
    - upon running an etl load, the topics channel page should be automatically populated with topics (since we did the static load via migration)


### Additional Context
On existing live environments like production and rc, the addition of the static fixture load will be ignored (since those migrations have already run) - on new environments this conveniently preloads the required data for us.
